### PR TITLE
Add a js implementation of `wasm-ext`.

### PR DESCRIPTION
**Pull request by expenses**
This PR just copies the implementation of the web socket transport we use for the substrate browser node to libp2p and binds to it via wasm-bindgen.

### CODE REVIEW

**Comment by tomaka** [comment]
One problem is that you're assuming websockets or even browser. This is a generic WebAssembly transport.

**Comment by expenses** [comment]
Right, so we should write js bindings to some things (`multiaddr_to_ws` etc) but not others? It might be better to plug in the implementation into `browser-utils` in substrate then.

**Comment by tomaka** [comment]
IMO we should instead add a `websocket_transport` feature to `libp2p-wasm-ext` and provide this transport only if the feature is enabled.

Also, the `transport` function should be named `websocket_transport`.

**Comment by expenses** [comment]
Yep, something like 
```
#[cfg_attr(feature = "websocket_transport", wasm_bindgen(module = "/src/implementation.js"))]
#[cfg_attr(not(feature = "websocket_transport"), wasm_bindgen)]
```
perhaps?

**Comment by tomaka** [comment]
Can you not split the `extern` block in two?

**Comment by expenses** [comment]
That might also work

File: transports/wasm-ext/src/lib.rs [hunk 1]
```diff
@@ -126,19 +123,20 @@ pub mod ffi {
         #[wasm_bindgen(method, getter)]
         pub fn local_addr(this: &ConnectionEvent) -> String;
     }
+
+    #[cfg(feature = "websockets")]
+    #[wasm_bindgen(module = "/src/websockets.js")]
+    extern "C" {
+        /// Returns a `Transport` implemented using websockets.
+        pub fn wasm_transport() -> Transport;
```
Comment (tomaka) [inline transports/wasm-ext/src/lib.rs]:
```suggestion
        pub fn websockets_transport() -> Transport;
```

Comment (expenses) [reply transports/wasm-ext/src/lib.rs]: Done

File: transports/wasm-ext/Cargo.toml [hunk 2]
```diff
@@ -16,3 +16,6 @@ libp2p-core = { version = "0.16.0", path = "../../core" }
 parity-send-wrapper = "0.1.0"
 wasm-bindgen = "0.2.42"
 wasm-bindgen-futures = "0.4.4"
+
+[features]
+websockets = []
```
Comment (tomaka) [inline transports/wasm-ext/Cargo.toml]:
```suggestion
websocket = []
```

File: transports/wasm-ext/src/lib.rs [hunk 3]
```diff
@@ -123,6 +123,13 @@ pub mod ffi {
         #[wasm_bindgen(method, getter)]
         pub fn local_addr(this: &ConnectionEvent) -> String;
     }
+
+    #[cfg(feature = "websockets")]
```
Comment (tomaka) [inline transports/wasm-ext/src/lib.rs]:
```suggestion
    #[cfg(feature = "websocket")]
```

**Comment by tomaka** [comment]
Ugh, obviously I forgot to merge before the 0.16.1 PR.

**Pull request merged**